examples/js/pmrem/PMREMGenerator.js

```diff
@@ -21,11 +19,13 @@ THREE.PMREMGenerator = ( function () {
 	scene.add( planeMesh );
 	scene.add( camera );
 
-	var PMREMGenerator = function ( sourceTexture, samplesPerLevel, resolution ) {
+	var PMREMGenerator = function ( sourceTexture, params = {} ) {
 
 		this.sourceTexture = sourceTexture;
-		this.resolution = ( resolution !== undefined ) ? resolution : 256; // NODE: 256 is currently hard coded in the glsl code for performance reasons
-		this.samplesPerLevel = ( samplesPerLevel !== undefined ) ? samplesPerLevel : 32;
+		this.sourceResolution = ( params.sourceResolution !== undefined ) ? params.sourceResolution : 512;
+		this.targetResolution = ( params.targetResolution !== undefined ) ? params.targetResolution : 256; // NODE: 256 is currently hard coded in the glsl code for performance reasons
+		this.samplesPerLevel = ( params.samplesPerLevel !== undefined ) ? params.samplesPerLevel : 32;
+		this.useImportanceSampling = ( params.useImportanceSampling !== undefined ) ? params.useImportanceSampling : false;
```

Comment:
`new PMREMGenerator({ useImportanceSampling: false })` would not use importance sampling in this case which would be unexpected

Reply:
Regard this style, I believe Three.JS does not use objects passed in as parameters, rather it uses ordered arguments.  The reason for this is that in generally objects used as parameter sets on a regular basis , because they are memory allocations on the heap, introduces GC churn.  Where as order arguments I believe are just stack based instead and thus do not impact GC.

Reply:
Makes sense! Maybe a breaking API change would be useful when/if moving to importance sampling method for consistency with core (1 object allocation won't add to overhead in the PMREM case) -- that being said, `useImportanceSampling: false` should use random sampling (use case, me flipping a boolean testing cubemaps back to back with importance vs random sampling)